### PR TITLE
add Kyocera address book module

### PR DESCRIPTION
**Pull request by ajm4n**
Added a module that exploits [cve-2022-1026](https://github.com/ac3lives/kyocera-cve-2022-1026)

### CODE REVIEW

File: modules/auxiliary/scanner/http/kyocera_addressbook_credential_gather.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        print_error("Failed to create address book enumeration on #{ip}")
+        print_status("Full HTML response: #{res.body}") # capture the HTML response
+      end
+    else
+      print_error("No response received from #{ip}")
+    end
+  end
+
+  def extract_enum_id(body)
+    xml_doc = Nokogiri::XML(body)
```
Comment (Chocapikk) [inline modules/auxiliary/scanner/http/kyocera_addressbook_credential_gather.rb:93]:
```suggestion
    xml_doc = res.get_xml_document
```

File: modules/auxiliary/scanner/http/kyocera_addressbook_credential_gather.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      print_status("Full response body: #{res.body}")
+
+      # Check if there's a redirection
+      if res.headers['Location']
+        print_status("Redirected to: #{res.headers['Location']}")
+      end
+
+      if res.code == 200
+        print_good("Enumeration creation successful on #{ip}")
+        enum_id = extract_enum_id(res.body)
```
Comment (Chocapikk) [inline modules/auxiliary/scanner/http/kyocera_addressbook_credential_gather.rb:73]:
```suggestion
        enum_id = extract_enum_id(res)
```

File: modules/auxiliary/scanner/http/kyocera_addressbook_credential_gather.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      else
+        print_error("Failed to create address book enumeration on #{ip}")
+        print_status("Full HTML response: #{res.body}") # capture the HTML response
+      end
+    else
+      print_error("No response received from #{ip}")
+    end
+  end
+
+  def extract_enum_id(body)
```
Comment (Chocapikk) [inline modules/auxiliary/scanner/http/kyocera_addressbook_credential_gather.rb:92]:
```suggestion
  def extract_enum_id(res)
```

**Comment by bwatters-r7** [comment]
@ajm4n Thank you so much for the PR!  I was taking a look at it, but before we get too much further, this PR is coming from your master branch, and that causes complications.  I'm going to mark this as a `PR from master branch`
As part of that process, github automations will close this PR.  That does not mean we do not want this PR; it just means it needs to come from a non-master branch on your side.  There will be instructions in the automated message.  If you have any questions, please reach out.
Thanks again, and we look forward to working with you!

**Pull request closed without merging**